playback/src/audio_backend/alsa.rs

```diff
@@ -1,20 +1,66 @@
 use super::{Open, Sink};
-use alsa::{Direction, Error, ValueOr};
 use alsa::device_name::HintIter;
-use std::ffi::{CStr, CString};
 use alsa::pcm::{Access, Format, HwParams, PCM};
+use alsa::{Direction, Error, ValueOr};
+use std::env;
+use std::ffi::CString;
 use std::io;
 use std::process::exit;
 
-
 pub struct AlsaSink(Option<PCM>, String);
 
 fn list_outputs() {
     for t in &["pcm", "ctl", "rawmidi", "timer", "seq", "hwdep"] {
-     println!("{} devices:", t);
-     let i = HintIter::new(None, &*CString::new(*t).unwrap()).unwrap();
-     for a in i { println!("  {:?}", a) }
- }
+        println!("{} devices:", t);
+        let i = HintIter::new(None, &*CString::new(*t).unwrap()).unwrap();
+        for a in i {
+            println!("  {:?}", a)
+        }
+    }
+}
+
+fn open_device(dev_name: &str) -> Result<(PCM), Box<Error>> {
+    let pcm = PCM::new(dev_name, Direction::Playback, false)?;
+    // http://www.linuxjournal.com/article/6735?page=0,1#N0x19ab2890.0x19ba78d8
+    // latency = period_size * periods / (rate * bytes_per_frame)
+    // For 16 Bit stereo data, one frame has a length of four bytes.
+    // 500ms  = buffer_size / (44100 * 4)
+    // buffer_size = 0.5 * 44100 = 22050 frames
+    {
+        // Set hardware parameters: 44100 Hz / Stereo / 16 bit
+        let hwp = HwParams::any(&pcm)?;
+
+        hwp.set_access(Access::RWInterleaved)?;
+        hwp.set_format(Format::s16())?;
+        hwp.set_rate(44100, ValueOr::Nearest)?;
+        hwp.set_channels(2)?;
+        // hwp.set_period_size_near(256, ValueOr::Nearest)?;
+        hwp.set_buffer_size_near(11025 * 2)?; // ~ 0.25 x 2 s latency
+
+        pcm.hw_params(&hwp)?;
+    }
+
+    // Additional software paramters + check
+    if env::var("LIBRESPOT_DEBUG").is_ok() {
```

Comment:
why not just use `debug!` macros?

Reply:
Either with `trace!` or `debug!` it would then show up on every `sink.open` when run with `--verbose`. Thought that might be a bit to much? But then again, don't know if this information is still required?

Reply:
Don't think it would be too much - people running in debug mode know what they're in for.

Reply:
Then there is always the added overhead of querying these parameters each time irregardless of if it is running in debug mode or not. I think I'll just drop this section. It proved useful when I was twiddling around with the parameters getting it to work. What do you think?